main.tf

```diff
@@ -18,11 +19,37 @@ resource "azurerm_public_ip" "landing_zone_vnet_pip_bastion" {
   sku                 = "Standard"
 }
 
+# Azure Bastion Subnet
+resource "azurerm_subnet" "AzureBastionSubnet" {
+  name                 = "AzureBastionSubnet" # Mandatory name - DO NOT CHANGE!!!
+  resource_group_name  = var.rg_connectivity_name
+  virtual_network_name = var.bastion_vnet_name
+  address_prefixes     = var.bastion_subnet_address
+}
+
+# Azure Bastion NSG
+resource "azurerm_network_security_group" "Bastion" {
+  name                = var.bastion_nsg_name
+  location            = var.location
+  resource_group_name = var.rg_connectivity_name
+  
+  depends_on = [
+    azurerm_subnet.AzureBastionSubnet
+  ]
+
+}
+
 ### Associate 
 # NSG association not possible for GatewaySubnet
 #
 resource "azurerm_subnet_network_security_group_association" "landing_zone_nsg_association_bastion" {
-  depends_on = [azurerm_network_security_rule.landing_zone_nsg_rule_bastion_0001,
+  subnet_id                 = azurerm_subnet.AzureBastionSubnet.id
+  network_security_group_id = azurerm_network_security_group.Bastion.id
+
+  depends_on = [
+    azurerm_network_security_group.Bastion,
+    azurerm_subnet.AzureBastionSubnet,
```

Comment:
You dont need to depend on azurerm_subnet.AzureBastionSubnet, when you use the output of a module as an input like for subnet_id = azurerm_subnet.AzureBastionSubnet.id. So you can delete all depends on, also the nsg rules because they depend on nsg